pandas/tools/pivot.py

```diff
@@ -106,6 +106,10 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
         else:
             values_multi = False
             values = [values]
+        # GH14938 Make sure values are in data
+        for i in values:
+            if i not in data:
+                raise KeyError(i)
```

Comment:
IMHO, it's better to be consistent with behavior on the other arguments. For example, with `groupby`, if you do this:
```python
df=pd.DataFrame({'a' : [i % 3 for i in range(10)], 'b': [i % 2 for i in range(10)], 'c': np.random.randn(10)})
df.groupby(['y','z']).sum()
```
The `KeyError` is raised on the value of `'y'`, not on `'z'`.  So I think the error raised on the `values` argument should be consistent.  Otherwise it begs the question why we don't show _all_ of the `KeyError` issues on other arguments for many other functions.